### ISSUE TITLE
`no-unexpected-multiline` false positive (eslint 2.0.0)

### ISSUE THREAD

**Issue by feross** [eslint/eslint#5148]
Using the latest eslint (2.0.0-rc.0).

Rules:

```json
{
  "rules": {
    "no-unexpected-multiline": 2
  }
}
```

Code:

```js
(
  function () {}
)[1]
```

Results in the following false positive:

```
  3:2  error  Unexpected newline between object and [ of property access  no-unexpected-multiline
```

This does not trigger an error in eslint v1.10.3.

**Comment by eslintbot** [comment]
@feross Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
Looks like a bug. Do you want to submit a PR for it?

**Comment by feross** [comment]
Sorry, wish I could, but I don't have time at the moment.

**Comment by feross** [comment]
Actually, I'm kinda curious what the cause is, so I'll take a quick look.

**Comment by feross** [comment]
Think I figured it out :+1:

**Comment by feross** [comment]
PR sent here: https://github.com/eslint/eslint/pull/5156
